### PR TITLE
[core] Deprecate enumerated properties

### PR DESCRIPTION
**Pull request by oowekyala**
Refs #1432

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">This changeset introduces 0 new violations and 0 new errors,<br />
removes 0 violations and 0 errors. <a href="http://chunk.io/pmd/1de5129cf8d04bd4bcc1202004b171aa/diff/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AvoidUsingHardCodedIPRule.java [hunk 1]
```diff
@@ -35,10 +39,10 @@
     }
 
 
-    public static final EnumeratedMultiProperty<String> CHECK_ADDRESS_TYPES_DESCRIPTOR = new EnumeratedMultiProperty<>(
-        "checkAddressTypes", "Check for IP address types.", ADDRESSES_TO_CHECK,
-        Arrays.asList(IPV4, IPV6, IPV4_MAPPED_IPV6),
-        String.class, 2.0f);
+    private static final PropertyDescriptor<List<String>> CHECK_ADDRESS_TYPES_DESCRIPTOR =
```
Comment (adangel) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AvoidUsingHardCodedIPRule.java]: I hope, it's ok to change this from public to private and change the type? I guess, nobody anyway directly used this property programmatically...

Comment (adangel) [reply pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AvoidUsingHardCodedIPRule.java]: I think, we should keep it public...

**Comment by adangel** [comment]
I'll merge this now. There are two changes, we should think about:
* https://github.com/pmd/pmd/pull/1505/files#diff-819361d88a860dde8f8a37fc3d8a9359R42
* https://github.com/pmd/pmd/pull/1505/files#diff-86e05550f73473551b6a7d196ac683e6R49

These change the type of public properties from `EnumeratedMultiProperty<String>` to `PropertyDescriptor<List<String>>` - which is a API incompatible change.
We could say, that this is tolerable for properties, that are defined and used by only one rule.
Wdyt?

**Comment by oowekyala** [comment]
Yeah I think for the moment, property descriptors for individual rules should be part of their implementation and so should be hidden. We could probably go so far as say that concrete rule classes are also not part of our published API ?  We'd anyway need to mention that on the release notes though

**Pull request merged**